zebra-consensus/src/primitives/groth16/tests.rs

```diff
@@ -0,0 +1,84 @@
+//! Tests for redjubjub signature verification
+
+use super::*;
+
+use std::time::Duration;
+
+use color_eyre::eyre::{eyre, Result};
+use futures::stream::{FuturesUnordered, StreamExt};
+use tower::ServiceExt;
+use tower_batch::Batch;
+
+async fn sign_and_verify<V>(mut verifier: V, n: usize) -> Result<(), V::Error>
+where
+    V: Service<Item, Response = ()>,
+{
+    let rng = thread_rng();
+    let mut results = FuturesUnordered::new();
+    for i in 0..n {
+        let span = tracing::trace_span!("sig", i);
+        let msg = b"BatchVerifyTest";
+
+        match i % 2 {
+            0 => {
+                let sk = SigningKey::<SpendAuth>::new(rng);
+                let vk = VerificationKey::from(&sk);
+                let sig = sk.sign(rng, &msg[..]);
+                verifier.ready_and().await?;
+                results.push(span.in_scope(|| verifier.call((vk.into(), sig, msg).into())))
+            }
+            1 => {
+                let sk = SigningKey::<Binding>::new(rng);
+                let vk = VerificationKey::from(&sk);
+                let sig = sk.sign(rng, &msg[..]);
+                verifier.ready_and().await?;
+                results.push(span.in_scope(|| verifier.call((vk.into(), sig, msg).into())))
+            }
+            _ => panic!(),
+        }
+    }
+
+    while let Some(result) = results.next().await {
+        result?;
+    }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn batch_flushes_on_max_items_test() -> Result<()> {
+    batch_flushes_on_max_items().await
+}
+
+#[spandoc::spandoc]
```

Comment:
since you have a `#[spandoc]` attribute here, it might be a cool idea to add a `/// SPANDOC:` comment before the main line in the test where it can potentially fail, so we get a nice description of what it was doing in the error message

Reply:
Oh man, this got committed when it was not updated away from redjubjub yet, but I should include the better `SPANDOC:` comment either way